proto/reaction.proto

```diff
@@ -319,6 +379,21 @@ message ReactionSetup {
   string automation_platform = 3;
   // Raw automation code or synthetic recipe definition.
   map<string, Data> automation_code = 4;
+  message ReactionEnvironment {
+    enum ReactionEnvironmentType {
+    UNSPECIFIED = 0;
+    CUSTOM = 1;
+    FUME_HOOD = 2;
+    BENCH_TOP = 3;
+    GLOVE_BOX = 4;
+    GLOVE_BAG = 5;
+    }
+    // Specify reaction environment.
+    ReactionEnvironmentType type = 1;
+    // Specify environment details, e.g., "nitrogen-filled" for GLOVE_BOX.
+    string details = 2;
```

Comment:
Since `environment` is not repeated, we should follow the conventions used above (see `vessel_material` and `vessel_material_details` for a good example) like so:

```
message ReactionEnvironment {
  enum ReactionEnvironmentType {
    UNSPECIFIED = 0;
    CUSTOM = 1;
    FUME_HOOD = 2;
    BENCH_TOP = 3;
    GLOVE_BOX = 4;
    GLOVE_BAG = 5;
    }
}
ReactionEnvironment.ReactionEnvironmentType environment = 5;
string environment_details = 6;

```

Reply:
Got it, thanks!